integrations/consul/documentation.yaml

```diff
@@ -0,0 +1,89 @@
+exporter_type: included
+app_name_short: Consul
+app_name: {{app_name_short}}
+app_site_name: Consul
+app_site_url: https://www.consul.io/
+exporter_name: the Consul exporter
+exporter_pkg_name: consul
+exporter_repo_url: https://developer.hashicorp.com/consul/docs/agent/config/config-files#telemetry-parameters
+dashboard_available: true
+minimum_exporter_version: "1.13.3"
+multiple_dashboards: false
+dashboard_display_name: {{app_name_short}} Prometheus Overview
+config_mods: |
+  apiVersion: v1
+  kind: ConfigMap
+  metadata:
+    name: consul-telemetry
+  data:
+  + metrics.hcl: |
+  +   telemetry {
+  +     prometheus_retention_time = "60s"
+  +     disable_hostname = true
+  +   }
+  ---
+  apiVersion: apps/v1
+  kind: StatefulSet
+  metadata:
+    name: consul
+  spec:
+    serviceName: consul
+    selector:
+      matchLabels:
+  +     app.kubernetes.io/name: consul
+    template:
+      metadata:
+        labels:
+  +       app.kubernetes.io/name: consul
+      spec:
+        containers:
+        - name: consul
+          image: consul:1.13.3
+          ports:
+          - containerPort: 8500
+            name: consul
+            protocol: TCP
+  +       volumeMounts:
+  +       - mountPath: /consul/config/metrics.hcl
+  +         subPath: metrics.hcl
+  +         name: consul-telemetry
+  +     volumes:
+  +     - name: consul-telemetry
+  +       configMap:
+  +         name: consul-telemetry
+  +         items:
+  +         - key: metrics.hcl
+  +           path: metrics.hcl
+podmonitoring_config: |
+  apiVersion: monitoring.googleapis.com/v1
+  kind: PodMonitoring
+  metadata:
+    name: consul
+    labels:
+      app.kubernetes.io/name: consul
+      app.kubernetes.io/part-of: google-cloud-managed-prometheus
+  spec:
+    endpoints:
+  + - port: 8500
+      scheme: http
+      interval: 30s
+  +   path: /v1/agent/metrics
+  +   params:
+  +     format: 
+  +       - prometheus
+    selector:
+      matchLabels:
+  +     app.kubernetes.io/name: consul
```

Comment:
We've been using "+"s to indicate the delta to an existing config to add telemetry and monitoring, so this makes sense under config_mod, but we shouldn't need to have "+"s in podmonitoring_config since we don't expect anyone to have an existing podmonitoring config they are modifying.